client/profile-wizard/steps/business-details/flows/selective-bundle/selective-extensions-bundle/index.js

```diff
@@ -263,11 +268,16 @@ export const SelectiveExtensionsBundle = ( {
 
 	useEffect( () => {
 		if ( ! isInstallingActivating ) {
-			setInstallExtensionOptions(
-				createInstallExtensionOptions( installableExtensions )
+			setInstallExtensionOptions( ( currInstallExtensionOptions ) =>
+				createInstallExtensionOptions( {
+					installableExtensions,
+					prevInstallExtensionOptions: currInstallExtensionOptions,
+				} )
 			);
 		}
-	}, [ installableExtensions, isInstallingActivating ] );
```

Comment:
The root cause is the `isInstallingActivating` will be changed after the user clicks the "Continue" button and this function call will reset the options state.